src/uglify/serialization.js

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable
+  no-new-func
+*/
+import espree from 'espree';
```

Comment:
Yeah reread this whole file and while it's a neat trick, I think simply catching the error and warning that you can't use parallel if you provide your own functions is probably the best idea.

Or even providing a generic handler, and post processing the results so you don't have to serialize the regex or function.